File: Python/백준/Platinum/15907. Acka의 리듬 세상/Acka의 리듬 세상.py
SIZE = 2000001

is_prime = [True] * SIZE
for i in range(2, int(SIZE**0.5)+1):
    if not is_prime[i]: continue
    for j in range(i*i, SIZE, i):
        is_prime[j] = False

N = int(input())
arr = [*map(int,input().split())]

ans = 0
for i in range(2,SIZE):
    if not is_prime[i]: continue
    if ans * i > arr[-1]: break
    data = dict()
    for num in arr:
        tmp = num % i
        data[tmp] = data.get(tmp, 0) + 1
    ans = max(ans, *data.values())

print(ans)

File: Go/백준/Platinum/15907. Acka의 리듬 세상/Acka의 리듬 세상.go
package main

import (
	"bufio"
	"math"
	"os"
	"strconv"
)

const SIZE = 2000001

func getPrime() []bool {
	isPrime := make([]bool, SIZE)
	for i := 2; i < int(math.Sqrt(SIZE))+1; i++ {
		if isPrime[i] {
			continue
		}
		for j := i * i; j < SIZE; j += i {
			isPrime[j] = true
		}
	}
	return isPrime
}

func main() {
	bs.Split(bufio.ScanWords)
	defer bw.Flush()

	isPrime := getPrime()

	N := input()
	arr := make([]int, N)

	for i := 0; i < N; i++ {
		arr[i] = input()
	}
	ans := 0
	for i := 2; i < SIZE; i++ {
		if isPrime[i] {
			continue
		}
		if ans*i > arr[N-1] {
			break
		}
		data := make(map[int]int)
		for j := 0; j < N; j++ {
			tmp := arr[j] % i
			data[tmp]++
		}
		for _, v := range data {
			ans = max(ans, v)
		}
	}

	bw.WriteString(strconv.Itoa(ans))
}

var (
	bs = bufio.NewScanner(os.Stdin)
	bw = bufio.NewWriter(os.Stdout)
)

func input() int {
	bs.Scan()
	num, _ := strconv.Atoi(bs.Text())
	return num
}
